src/components/password/password.stories.mdx

```diff
@@ -0,0 +1,150 @@
+
+## Contents
+
+- [Quick Start](#quick-start)
+- [Examples](#examples)
+- [Props](#props)
+
+## Quick Start
+
+```javascript
+import Password from "carbon-react/lib/components/password";
+```
+
+## Examples
+
+### Default
+
+Please ensure that the input type is set to `password` on form submission, this will ensure browsers do not remember
+the inputs content. We have set the `autocomplete` attribute to `off` on the component, but this is not widely 
+supported across different browsers and can have unpredictable results.
+
+<Canvas>
+  <Story name="default" story={stories.Default} />
+</Canvas>
+
+
+### with inputHint
+
+<Canvas>
+  <Story name="with inputHint" story={stories.InputHint} />
+</Canvas>
+
+### Character counter
+
+<Canvas>
+  <Story name="with character counter" story={stories.CharacterCounter} />
+</Canvas>
+
+
+### Prefix
+
+<Canvas>
+  <Story name="with prefix" story={stories.Prefix} />
+</Canvas>
+
+
+### Sizes
+
+<Canvas>
+  <Story name="with sizes" story={stories.Sizes} />
+</Canvas>
+
+### Margins
+
+<Canvas>
+  <Story name="with margins" story={stories.Margins} />
+</Canvas>
+
+### Disabled
+
+<Canvas>
+  <Story name="disabled" story={stories.Disabled} />
+</Canvas>
+
+### ReadOnly
+
+<Canvas>
+  <Story name="readOnly" story={stories.ReadOnly} />
+</Canvas>
+
+### AutoFocus
+
+<Canvas>
+  <Story name="with autoFocus" story={stories.AutoFocus} />
+</Canvas>
+
+### With labelInline
+
+<Canvas>
+  <Story name="with labelInline" story={stories.WithLabelInline} />
+</Canvas>
+
+### With label Align
+
+<Canvas>
+  <Story name="with labelAlign" story={stories.WithLabelAlign} />
+</Canvas>
+
+### With custom labelWidth and inputWidth
+
+<Canvas>
+  <Story name="with custom labelWidth and inputWidth" story={stories.WithCustomLabelWidthAndInputWidth} />
+</Canvas>
+
+### With custom maxWidth
+
+<Canvas>
+  <Story name="with custom maxWidth" story={stories.WithCustomMaxWidth} />
+</Canvas>
+
+### With fieldHelp
+
+<Canvas>
+  <Story name="with fieldHelp" story={stories.WithFieldHelp} />
+</Canvas>
+
+### With labelHelp
+
+<Canvas>
+  <Story name="with labelHelp" story={stories.WithLabelHelp} />
+</Canvas>
+
+### With required
+
+<Canvas>
+  <Story name="required" story={stories.WithRequired} />
+</Canvas>
+
+
+### Validations
+
+Validation status can be set by passing `error`, `warning` or `info` prop to the component.
+
+Passing a string to these props will display a properly colored border along with a validation icon and tooltip - string value will be displayed as the tooltip message.
+
+Passing a boolean to these props will display only a properly colored border.
+
+For more information check our [Validations](?path=/docs/documentation-validations--string-validation "Validations") documentation page.
+
+## Props
+
+### Password
+
+Due to the `Textbox` component being used internally by the `Password` component all the props of `Textbox` can be applied to `Password` as well.
```

Comment:
In my opinion, we should be surfacing `type` because consumers will want to enter their own mechanisms to change the `type` back to `password` on submission etc.